code/game/machinery/computer/singulo_monitor.dm

```diff
@@ -0,0 +1,187 @@
+		return
+	ui_interact(user)
+
+/obj/machinery/computer/singulo_monitor/ui_state(mob/user)
+	return GLOB.default_state
+
+/obj/machinery/computer/singulo_monitor/ui_interact(mob/user, datum/tgui/ui = null)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "SingularityMonitor", name)
+		ui.open()
+
+/obj/machinery/computer/singulo_monitor/ui_data(mob/user)
+	var/list/data = list()
+
+	if(istype(active))
+		var/turf/T = get_turf(active)
+		// If we somehow dissipate during this proc, handle it somewhat
+		if(!T)
+			active = null
+			refresh()
+			return
+
+		data["active"] = TRUE
+		data["singulo_stage"] = (active.current_size + 1) / 2
+		data["singulo_potential_stage"] = (active.allowed_size + 1) / 2
+		data["singulo_energy"] = active.energy
+
+		switch(active.allowed_size)
+			if(STAGE_ONE)
+				data["singulo_high"] = STAGE_TWO_THRESHOLD
+				data["singulo_low"] = 0
+			if(STAGE_TWO)
+				data["singulo_high"] = STAGE_THREE_THRESHOLD
+				data["singulo_low"] = STAGE_TWO_THRESHOLD
+			if(STAGE_THREE)
+				data["singulo_high"] = STAGE_FOUR_THRESHOLD
+				data["singulo_low"] = STAGE_THREE_THRESHOLD
+			if(STAGE_FOUR)
+				data["singulo_high"] = STAGE_FIVE_THRESHOLD
+				data["singulo_low"] = STAGE_FOUR_THRESHOLD
+			else
+				data["singulo_high"] = STAGE_SIX_THRESHOLD
+				data["singulo_low"] = STAGE_FIVE_THRESHOLD
+
+		var/list/generators = list()
+		var/index = 1
+		for(var/obj/machinery/field/generator/generator in field_gens)
+			generators.Add(list(list(
+				"charge" = generator.energy,
+				"gen_index" = index++
+			)))
+		data["generators"] = generators
+
+	else
+		var/list/singulos = list()
+		for(var/obj/singularity/S in singularities)
+			var/area/A = get_area(S)
+			if(!A)
+				continue
+
+			singulos.Add(list(list(
+				"area_name" = A.name,
+				"energy" = S.energy,
+				"stage" = (S.current_size + 1) / 2,
+				"singularity_id" = S.singulo_id
+			)))
+
+		data["active"] = FALSE
+		data["singularities"] = singulos
+
+	return data
+
+/**
+  * Supermatter List Refresher
+  *
+  * This proc loops through the list of supermatters in the atmos SS and adds them to this console's cache list
+  */
+/obj/machinery/computer/singulo_monitor/proc/refresh()
+	singularities = list()
+	var/turf/T = get_turf(ui_host()) // Get the UI host incase this ever turned into a singularity monitoring module for AIs to use or something
+	if(!T)
+		return
+	for(var/obj/singularity/S in GLOB.singularities)
+		// not within coverage, not on a tile, not a gravitational singularity
+		if(S.type != /obj/singularity || !(is_station_level(S.z) || is_mining_level(S.z) || atoms_share_level(S, T)))
+			continue
+		singularities.Add(S)
+
+	if(!(active in singularities))
+		active = null
+
+/obj/machinery/computer/singulo_monitor/proc/send_alerts()
+	// Breach alerts
+	if(length(field_gens) && (active.current_size > (last_size + 2) || active.current_size >= STAGE_FIVE))// We should only see a singulo grow 2 stages at once when breaching containment.
```

Comment:
```suggestion
	if(active.current_size > (last_size + 2) || active.current_size >= STAGE_FIVE)// We should only see a singulo grow 2 stages at once when breaching containment.
```

Why check to see if field gens are present to announce if a singlo is loose? If none are present, its likely that the singlo ate them, or an evildoer intentionally didn't use them.

Reply:
There are two separate alerts, one for singuloose and another for loss of the containment field. Reason being the containment field can be broken and repaired for a level 1 singulo.

Reply:
Oh disregard what I said. This is so it doesn't give alerts for uncontained singularities. Otherwise it would give constant alerts

Reply:
Oh that's very clever. Thank you!